### PR TITLE
feat: unlock app with biometrics (WPB-4696)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4696" title="WPB-4696" target="_blank">StoryWPB-4696</a>  [Android] Use Biometric for unlocking the app if available  
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Description

This PR will allow the user to unlock his device using biometrics (fingerprint or FaceId). 

The biometrics prompt dialog will be displayed if you have your fingerprint or faceId configured in device otherwise regular app lock screen will be shown.

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

- Make sure that you have 
- Enable app lock
- Put the app in background for 60s
- Open the app again, you should see biometrics prompt dialog.

### Attachments (Optional)

| Samsung S22 Fingerprint  | Samsung S22 FaceId |
| ------------- | ------------- |
| Wire 2023-10-12 at 2_59 PM  | Wire 2023-10-12 at 2_59 PM2  |

| Pixel 6 Fingerprint  | 
| ------------- | 
| Wire 2023-10-12 at 4_02 PM  | 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/biomitric/BiometricPromptUtils.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        }
+        return BiometricPrompt(activity, executor, callback)
+    }
+
+    fun createPromptInfo(): BiometricPrompt.PromptInfo =
+        BiometricPrompt.PromptInfo.Builder().apply {
+            setTitle("Verify that it's you")
+            setSubtitle("To unlock Wire")
+            setConfirmationRequired(false)
+            setNegativeButtonText("Use passcode")
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/biomitric/BiometricPromptUtils.kt]: Localisation in these Strings!

**Review by vitorhugods — CHANGES_REQUESTED** [review]
LGTM, just wonder about having some strings localised

Exciting to have app lock :)

File: app/src/main/kotlin/com/wire/android/biomitric/BiometricPromptUtils.kt (hunk 1 above)
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/biomitric/BiometricPromptUtils.kt]: Oh! yeah

**Review by vitorhugods — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [1269](https://10.10.124.134/job/AR-build-pipeline/1269/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7815370)(7815370)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7815370)(7815370)-staging-compat.apk)

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [1306](https://10.10.124.134/job/AR-build-pipeline/1306/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7824567)(7824567)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7824567)(7824567)-staging-compat.apk)

**Review by saleniuk — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [1318](https://10.10.124.134/job/AR-build-pipeline/1318/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7847702)(7847702)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7847702)(7847702)-staging-compat.apk)

**Pull request merged**